Moco/exampleWalking/exampleWalking.m

```diff
@@ -0,0 +1,284 @@
+% Data are from a processed OpenCap trial of walking and contain coordinate
+% values after inverse kinematics.
+
+clear;
+
+% Load the Moco libraries
+import org.opensim.modeling.*;
+
+% ---------------------------------------------------------------------------
+% Set up a coordinate tracking problem where the goal is to minimize the
+% difference between provided and simulated coordinate values and speeds, 
+% as well as to minimize an effort cost (squared controls). Endpoint
+% constraints enforce periodicity of the coordinate values (except for
+% pelvis tx) and speeds, coordinate actuator controls, and muscle 
+% activations.
+
+
+% Define the optimal control problem
+% ==================================
+mocoTrackName = 'walking1';
+track = MocoTrack();
+track.setName(mocoTrackName);
+
+% Set the weights for the terms in the objective function. The values below were
+% obtained by trial and error.
+controlEffortWeight = 0.1;
+stateTrackingWeight = 1;
+
+% Set other parameters, including the initial and final times of the
+% simulation, the mesh interval, the relative size of the state bounds, and
+% the optimal force for the lumbar and upper limb coordinate actuators.
+initialTime = 0.07;
+finalTime = 1.45;
+meshInterval = 0.05;
+fractionExtraBoundSize = 0.1;
+coordinateActuatorsOptimalForce = 250;
+
+% Reference data for tracking problem
+tableProcessor = TableProcessor('walking1_videoAndMocap.mot');
+tableProcessor.append(TabOpLowPassFilter(6));
+tableProcessor.append(TabOpUseAbsoluteStateNames());
+
+% Load the model and increase the optimal force for the coordinate
+% actuators that control the lumbar and upper limbs.
+inModel = Model('../models/LaiArnoldModified2017_contacts.osim');
+setCoordinateActuatorsOptimalForce(inModel, coordinateActuatorsOptimalForce);
+
+% Process the model to weld the toe (mtp) joints, replace the muscles in
+% the model to use the DeGrooteFregly2016 muscles, and to use the implicit
+% mode for the muscles.
+modelProcessor = ModelProcessor(inModel);
+jointsToWeld = StdVectorString();
+jointsToWeld.add('mtp_r');
+jointsToWeld.add('mtp_l');
+modelProcessor.append(ModOpReplaceJointsWithWelds(jointsToWeld));
+modelProcessor.append(ModOpReplaceMusclesWithDeGrooteFregly2016());
+modelProcessor.append(ModOpTendonComplianceDynamicsModeDGF('implicit')); 
+
+% Set the settings for the MocoTrack problem.
+track.setModel(modelProcessor);
+track.setStatesReference(tableProcessor);
+track.set_states_global_tracking_weight(stateTrackingWeight);
+track.set_allow_unused_references(true);
+track.set_track_reference_position_derivatives(true);
+track.set_apply_tracked_states_to_guess(true);
+track.set_initial_time(initialTime);
+track.set_final_time(finalTime);
+track.set_mesh_interval(meshInterval);
+study = track.initialize();
+problem = study.updProblem();
+
+
+% Goals
+% =====
+
+% Tracking
+% --------
+% Set different tracking weights for states (weights for states not 
+% explicitly set here have a default value of 1.0)
```

Comment:
said these were determined by trial and error in the squat example. would be good to have here too.